### DISCUSSION TITLE
Question about the Postprocessor  'PointValue '

### DISCUSSION THREAD

**Discussion by echo1115** [idaholab/moose#23237] [Q&A General] [answered]
Hi all,
As described in [https://mooseframework.inl.gov/source/postprocessors/PointValue.html](url) , when using `PointValue `, it will throw an error if the point being sampled lies outside of the domain. 

If we use it to get the variable at a point on the  surface of simulating object, there would be no mesh at the original location of the point because the mesh deforms during calculating, I want to konw whether this could cause an error or how to avoid it.

Thanks in advance!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You ll want to use the SideValueSampler, this will follow the mesh.
https://mooseframework.inl.gov/source/vectorpostprocessors/SideValueSampler.html

Otherwise you can try the use_displaced_mesh parameter but sampling on an external boundary is already iffy, you're depending on rounding errors placing the point inside the mesh

Guillaume

**Reply by echo1115** [reply]
@GiudGiud  Thanks for your reply. Acturally I want to get the value at specific point on the surface,  but the `SideValueSampler`  cannot set a point parameter. As for the `use_displaced_mesh` parameter in PointValue , does it means by setting it to 'true' then it will keep following the original point I set even the mesh deforms?

**Reply by GiudGiud** [reply]
see this explanation for the behavior of PPs under displaced meshes
https://github.com/idaholab/moose/discussions/22935#discussioncomment-4358095

**Reply by echo1115** [reply]
@GiudGiud Thanks.  So according to the explanation, to avoid the mesh displace casusing the point locates at unexist domain, I should set `use_displaced_mesh = false` , which is also the default option. Another question: 
For getting variables on surface by using PPs, whether only the nodal variable can be obtained but elemental variable can't?

**Reply by GiudGiud** [reply]
Elemental variables are not always defined on a boundary, they can be discontiuous there it depends on the variable type. 

So it might not be supported to sample them on boundaries, as devs have been a little too careful banning elemental variables instead of banning discontinuous ones

**Reply by echo1115** [reply]
Ok,thanks a lot for your kindly help! @GiudGiud
